### ISSUE TITLE
Remove logical XOR

### ISSUE THREAD

**Issue by rafaelsamenezes** [esbmc/esbmc#1132]
Is there any need for these expressions?

**Comment by fbrausse** [comment]
Not sure why you picked these, but from what I can see, Abs is also unused.

Implications are directly supported by the solvers. Personally, I find reasoning over those easier than over their classical counter part, which also is larger, AST-wise.

Sorry if off-topic: For the pointer-analysis it could be nice to have a `distinct2t` with a vector of sub-exprs.

**Comment by rafaelsamenezes** [comment]
> Not sure why you picked these, but from what I can see,

I am just adding support to more expressions at the interval analysis and got curious about them.

>  Abs is also unused.

 It is converted from irep1 (when calling abs, labs, etc...). Then migrate converts it to irep2

**Comment by rafaelsamenezes** [comment]
> For the pointer-analysis it could be nice to have a distinct2t with a vector of sub-exprs.

Could you clarify a bit (maybe in a new issue)? After the interval analysis I will take a look at the points-to analysis.

**Comment by fbrausse** [comment]
> > Abs is also unused.
> 
> It is converted from irep1 (when calling abs, labs, etc...). Then migrate converts it to irep2

Oh, indeed, sorry I missed that.

> > For the pointer-analysis it could be nice to have a distinct2t with a vector of sub-exprs.
> 
> Could you clarify a bit (maybe in a new issue)? After the interval analysis I will take a look at the points-to analysis.

Will do when I'm back working on it.

**Comment by rafaelsamenezes** [comment]
@fbrausse do you know any reason for the logical xor, isn't that equivalent to not_equal?

**Comment by fbrausse** [comment]
@rafaelsamenezes Yeah, it is. I don't know of any reason to keep it if its unused.

**Comment by mikhailramalho** [comment]
Is that a bitwise xor? Don't we use xor for integer/real encoding?

I remember we use the imply operator in the SMT backend, can you double check before removing it?

**Comment by rafaelsamenezes** [comment]
> Is that a bitwise xor? 

No, we have both the xor and bitwise_xor. 

> Don't we use xor for integer/real encoding?

~I don't think that we use any of them.~

Apparently we use the SMT interface `mk_xor` directly when we need. Without the use of xor2t/bitxor2t

> I remember we use the imply operator in the SMT backend

I will keep implication

**Comment by mikhailramalho** [comment]
Can you double check how a program like:

int a = 0, b = 0;
int c = a ^ b;

Is enconded with --ir?

**Comment by rafaelsamenezes** [comment]
The XOR is now used by the Python frontend.
